hedera-node/src/main/java/com/hedera/services/txns/token/TokenDeleteTransitionLogic.java

```diff
@@ -47,27 +46,41 @@
 
 	private final Function<TransactionBody, ResponseCodeEnum> SEMANTIC_CHECK = this::validate;
 
-	private final TokenStore store;
 	private final TransactionContext txnCtx;
+	private final TypedTokenStore tokenStore;
 
 	public TokenDeleteTransitionLogic(
-			TokenStore store,
-			TransactionContext txnCtx
+			final TransactionContext txnCtx,
+			final TypedTokenStore tokenStore
 	) {
-		this.store = store;
 		this.txnCtx = txnCtx;
+		this.tokenStore = tokenStore;
 	}
 
 	@Override
 	public void doStateTransition() {
-		try {
-			var op = txnCtx.accessor().getTxn().getTokenDeletion();
-			var outcome = store.delete(op.getToken());
-			txnCtx.setStatus((outcome == OK) ? SUCCESS : outcome);
-		} catch (Exception e) {
-			log.warn("Unhandled error while processing :: {}!", txnCtx.accessor().getSignedTxnWrapper(), e);
-			txnCtx.setStatus(FAIL_INVALID);
-		}
+
+		/* --- Translate from gRPC types --- */
+
+		final var op = txnCtx.accessor().getTxn().getTokenDeletion();
+
+		final var grpcTokenId = op.getToken();
+
+		/* --- Convert to model id --- */
+
+		final var targetTokenId = Id.fromGrpcToken(grpcTokenId);
+
+		/* --- Load the model object --- */
+
+		final var loadedToken = tokenStore.loadToken(targetTokenId);
+
+		/* --- Do the business logic --- */
+
+		loadedToken.delete();
+
+		/* --- Persist the updated model --- */
+
+		tokenStore.persistToken(loadedToken);
```

Comment:
nit: Lets remove all those empty lines. Conceptually it makes the code harder to read. We could have the occasional  empty line between the `comments`